bindings/python/convert.py

```diff
@@ -1,72 +1,189 @@
+        newdata["weight_map"] = newmap
+        json.dump(newdata, f)
+    local_filenames.append(index)
+
+    operations = [CommitOperationAdd(path_in_repo=local.split("/")[-1], path_or_fileobj=local) for local in local_filenames]
+
+    return operations
+
+
+def convert_single(model_id: str, folder: str) -> List["CommitOperationAdd"]:
+    sf_filename = "model.safetensors"
+    filename = hf_hub_download(repo_id=model_id, filename="pytorch_model.bin")
+    loaded = torch.load(filename)
+
+    local = os.path.join(folder, sf_filename)
+    shared = shared_pointers(loaded)
+    for shared_weights in shared:
+        for name in shared_weights[1:]:
+            loaded.pop(name)
+
+    # For tensors to be contiguous
+    loaded = {k: v.contiguous() for k, v in loaded.items()}
+
+    save_file(loaded, local, metadata={"format": "pt"})
+
+    check_file_size(local, filename)
+
+    operations = [CommitOperationAdd(path_in_repo=sf_filename, path_or_fileobj=local)]
+    return operations
+
+def check_final_model(model_id: str, folder: str):
+    config = hf_hub_download(repo_id=model_id, filename="config.json")
+    shutil.copy(config, os.path.join(folder, "config.json"))
+    config = AutoConfig.from_pretrained(folder)
+
+    _, pt_model = infer_framework_load_model(model_id, config)
+    _, sf_model = infer_framework_load_model(folder, config)
 
-        api = HfApi()
+    pt_model = pt_model
+    sf_model = sf_model
 
-        api.upload_file(
-            path_or_fileobj=local,
-            create_pr=True,
-            path_in_repo=local,
-            repo_id=model_id,
-        )
-    finally:
-        os.remove(local)
+    pt_params = pt_model.state_dict()
+    sf_params = sf_model.state_dict()
+
+    pt_shared = shared_pointers(pt_params)
+    sf_shared = shared_pointers(sf_params)
+    if pt_shared != sf_shared:
+        raise RuntimeError("The reconstructed model is wrong, shared tensors are different {shared_pt} != {shared_tf}")
+
+    sig = signature(pt_model.forward)
+    input_ids = torch.arange(10).unsqueeze(0)
+    pixel_values = torch.randn(1, 3, 224, 224)
+    input_values = torch.arange(1000).float().unsqueeze(0)
+    kwargs = {}
+    if "input_ids" in sig.parameters:
+        kwargs["input_ids"] = input_ids
+    if "decoder_input_ids" in sig.parameters:
+        kwargs["decoder_input_ids"] = input_ids
+    if "pixel_values" in sig.parameters:
+        kwargs["pixel_values"] = pixel_values
+    if "input_values" in sig.parameters:
+        kwargs["input_values"] = input_values
+    if "bbox" in sig.parameters:
+        kwargs["bbox"] = torch.zeros((1, 10, 4)).long()
+    if "image" in sig.parameters:
+        kwargs["image"] = pixel_values
+
+
+    if torch.cuda.is_available():
+        pt_model = pt_model.cuda()
+        sf_model = sf_model.cuda()
+        kwargs = {k: v.cuda() for k, v in kwargs.items()}
+
+    pt_logits = pt_model(**kwargs)[0]
+    sf_logits = sf_model(**kwargs)[0]
```

Comment:
I would suggest to test against all hidden states with `output_hidden_states=True` (whenever possible)

like
https://github.com/huggingface/transformers/blob/b047472650cba259621549ac27b18fd2066ce18e/src/transformers/commands/pt_to_tf.py#L292